cmd/operator-sdk/new/cmd.go

```diff
@@ -382,8 +384,16 @@ func getDeps() error {
 		if err := execProjCmd("dep", "ensure", "-v"); err != nil {
 			return err
 		}
+	case projutil.DepManagerGoMod:
+		log.Info("Running go mod ...")
+		if err := execProjCmd("go", "mod", "vendor", "-v"); err != nil {
+			return err
+		}
+		if err := execProjCmd("go", "mod", "tidy", "-v"); err != nil {
```

Comment:
I believe that `go mod tidy` should be run before `go mod vendor`, as tidy might edit the go.mod file

Reply:
I'd rather remove `go mod tidy` because it modifies `go.mod` and `go.sum`, changes which are then reverted by `go mod vendor`.

Reply:
I'm not sure that go mod vendor can revert changes to go.mod. From : https://github.com/thepudds/go-module-knobs/blob/master/README.md

> go mod vendor command
Resets the main module's vendor directory to include all packages needed to build and test all of the main module's packages based on the state of the go.mod files and Go source code.

It would be surprising that the vendor dir be different than the state recorded in the go.mod file as this file contains the cannonical truth on dependencies.

Reply:
Perhaps revert is not the right word. `go.mod` and `go.sum` are modified when you run `go mod tidy`, then changed again when you run `go mod vendor`. I assume its because of `tidy` functionality:

> The 'go mod tidy' command builds that view and then adds any missing module requirements and removes unnecessary ones.

While `vendor` just reads what's in the `vendor/` directory.

Also see the note in step 2 of [How to define a module](https://github.com/golang/go/wiki/Modules#how-to-define-a-module). We shouldn't run `go mod tidy` in `operator-sdk new`.